configure

```diff
@@ -113,10 +114,12 @@ command -v ${CC} 2>/dev/null 1>&2 || {
 	echo "${CC} not found: trying clang" 1>&2
 	echo "${CC} not found: trying clang" 1>&3
 	CC=clang
+	LINKER_SONAME_OPTION="-install_name"
```

Comment:
This shouldn't be conditional on the type of the C compiler (and the executable name is not a reliable way to detect the type anyways), but instead the type of the linker.

Only macOS' cctools will use `-install_name`, but clang can also be used with GNU binutils or lld.